File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    hist = []
    for i in range(len(s)):
        if s[i] == "(":
            hist.append(s[i])
        elif s[i] == ")":
            if len(hist) > 0:
                hist.pop()
            else:
                answer = False
    if len(hist) > 0:
        answer = False

    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include<vector>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    
    vector<char> check;
    if (s[0] == ')')
        return false;
    
    check.push_back(s[0]);
    for (int i = 1; i < s.size(); i++)
    {
        if (s[i] == ')')
        {
            if (check.size() > 0)
            {
                if (check.back() == '(')
                    check.pop_back();
                else
                    return false;
            }
            else
                return false;
        }
        else if (s[i] == '(')
            check.push_back(s[i]);
    }
    if (check.size() > 0)
        return false;

    return answer;
}